src/Mnemonic.js

```diff
@@ -136,6 +138,57 @@ export default class Mnemonic {
         );
     }
 
+    /**
+     * Converts a derivation path from string to an array of integers.
+     * Note that this expects precisely 5 components in the derivation path,
+     * as per BIP-44:
+     * `m / purpose' / coin_type' / account' / change / address_index`
+     * Takes into account `'` for hardening as per BIP-32,
+     * and does not prescribe which components should be hardened.
+     *
+     * @param {string} derivationPath    the derivation path in BIP-44 format,
+     *                                   e.g. "m/44'/60'/0'/0/0"
+     * @returns {Array<number>} to be used with PrivateKey#derive
+     */
+    calculateDerivationPathValues(derivationPath) {
+        // Parse the derivation path from string into values
+        const pattern = /m\/(\d+'?)\/(\d+'?)\/(\d+'?)\/(\d+'?)\/(\d+'?)/;
+        const matches = pattern.exec(derivationPath);
+        const values = new Array(5); // as Array<Number>;
+        if (matches) {
+            // Extract numbers and use apostrophe to select if is hardened
+            for (let i = 1; i <= 5; i++) {
+                let value = matches[i];
+                if (value.endsWith("'")) {
+                    value = value.substring(0, value.length - 1);
+                    values[i - 1] = parseInt(value, 10) | HARDENED_BIT;
+                } else {
+                    values[i - 1] = parseInt(value, 10);
+                }
+            }
+        }
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+        return values;
+    }
+
+    /**
+     * Common implementation for both `toStandardECDSAsecp256k1PrivateKey`
+     * functions.
+     *
+     * @param {string} passphrase  the passphrase used to protect the
+     *                             mnemonic, use "" for none
+     * @param {Array<number>} derivationPathValues derivation path as an
+     *                             integer array,
+     *                             see: `calculateDerivationPathValues`
+     * @returns a private key
+     */
+    async toStandardECDSAsecp256k1PrivateKeyImpl(
+        passphrase,
+        derivationPathValues,
+    ) {
+        return this.toEcdsaPrivateKey(passphrase, derivationPathValues);
```

Comment:
```suggestion
        return await this.toEcdsaPrivateKey(passphrase, derivationPathValues);
```

Reply:
@svetoslav-nikol0v Thanks - I have implemented both of the changes requested in [1708ea5](https://github.com/hashgraph/hedera-sdk-js/pull/2341/commits/1708ea56b9e6212529b698807c8ba9ae9f638e90)